winterfell/controllers/authController.js

```diff
@@ -57,13 +69,13 @@ module.exports = function (app) {
                 if (err) { return next(err); }
                 req.session.key = req.body.email;
                 req.session.userId = req.user._id;
-                return res.status(http.OK).send({redirect: '/'});
+                return res.redirect('/');
               });
             }
           );
         });
       } else { // User does exist!
-          res.status(http.BAD_REQUEST).send({error: httpErrors.USER_EXISTS});
+          res.redirect('/signup?error=EUSERNAMETAKEN');
```

Comment:
When we redirect to signup or login with an error message we should handle that on the front end JavaScript side. ,Anne look for the Parma and if it's there call the displayError message?